### DISCUSSION TITLE
GFX and xrdp_painter Q.

### DISCUSSION THREAD

**Discussion by J-W-C** [neutrinolabs/xrdp#3002] [General]
Hi all,

I noticed @Nexarian 's "Initial rfx progressive integration" committed 31 January to devel includes changes to xrdp/xrdp_painbter.c that add an alternate path in `xrdp_painter_send_dirty()`, one that avoids calling `libxrdp_send_bitmap()`, based on the value of client_info->gfx.

I'm wondering if future work along this line will also allow painter screens, like session login, to be sent using the original RemoteFX codec, if that's what a client indicates it wants?

/jwc

**Comment by matt335672** [comment]
Hi @J-W-C 

I don't think we've really thought about that as an option.

The login screen isn't performance intensive, so it doesn't really matter for that. That codepath is also used for the VNC and NeutrinoRDP backends though in GFX mode, so there could be some benefit for them.

**Reply by metalefty** [reply]
Same with @matt335672. The login screen isn't performance intensive.

**Comment by J-W-C** [comment]
Thanks guys.  I'll keep maintaining my fork for some people I work with who have an RDP client that needs it.

(It works but it doesn't really seem right to have a bunch of RFX codec-specific stuff in xrdp_painter_send_dirty() - working out which tiles are needed to cover the dirty rectangles, etc - when that's being done somewhere else for the Xorg server.)
